src/wallet/wallet.cpp

```diff
@@ -3960,6 +3963,8 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
     // Try to top up keypool. No-op if the wallet is locked.
     walletInstance->TopUpKeyPool();
 
+    LOCK(cs_main);
```

Comment:
Hmm, this is gonna conflict partially with #11281. The neccessary refactors seem somewhat unrelated, so I'd be fine with pushing the conflict resolution onto #11281, but if #11281 lands first this is gonna need a bit of tweaking...up to you if you want to do it now.